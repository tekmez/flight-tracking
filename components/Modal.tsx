import { Modal as RNModal, View, Platform } from "react-native";
import { BlurView } from "expo-blur";

type ModalProps = {
  visible: boolean;
  onClose: () => void;
  children: React.ReactNode;
};

const Modal = ({ visible, onClose, children }: ModalProps) => {
  return (
    <RNModal
      visible={visible}
      onRequestClose={onClose}
      transparent={true}
      animationType="fade"
    >
      <BlurView
        intensity={Platform.OS === "ios" ? 20 : 10}
        className="flex-1"
        tint="dark"
        experimentalBlurMethod="dimezisBlurView"
      >
        <View className="flex-1 justify-center items-center p-4 bg-[#37415199] ">
          {children}
        </View>
      </BlurView>
    </RNModal>
  );
};

export default Modal;
