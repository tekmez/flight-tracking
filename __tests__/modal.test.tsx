import Modal from "@/components/Modal";
import { render, fireEvent } from "@testing-library/react-native";
import { Text } from "react-native";

describe("Modal", () => {
  it("renders correctly", () => {
    const tree = render(
      <Modal visible={true} onClose={() => {}}>
        <Text>Hello</Text>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("onClose works correctly", () => {
    const mockFn = jest.fn();
    const modal = render(
      <Modal visible={true} onClose={mockFn}>
        <Text>Hello</Text>
      </Modal>
    );
    fireEvent.press(modal.getByText("Hello"));
    expect(mockFn).toHaveBeenCalled();
    modal.rerender(
      <Modal visible={false} onClose={mockFn}>
        <Text>Hello</Text>
      </Modal>
    );
    expect(modal.queryByText("Hello")).toBeNull();
  });
});
