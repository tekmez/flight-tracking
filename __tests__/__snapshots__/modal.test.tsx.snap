// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ],
      ]
    }
  >
    <ViewManagerAdapter_ExpoBlurView
      blurReductionFactor={4}
      experimentalBlurMethod="dimezisBlurView"
      intensity={20}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      tint="dark"
    />
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "backgroundColor": "#37415199",
          },
        ]
      }
    >
      <Text>
        Hello
      </Text>
    </View>
  </View>
</Modal>
`;
